web/soubory/program-nahled.js

```diff
@@ -0,0 +1,119 @@
+/**
+ * @param obalNahledu jquery element set element, ve kterém je náhled
+ * @param obalProgramu jquery element set element, ve kterém je program (kvůli
+ *  úpravě šířky při zobrazení náhledu)
+ * @param odkazy jquery element set odkazy na aktivity. Musí mít nastavený data
+ *  atribut program-nahled-id. Nejde o všechny odkazy, ale jen odkazy na
+ *  aktivity.
+ * @param prepinac jquery element set prvek, kterým se zapíná/vypíná zobrazení
+ *  náhledu.
+ */
+function programNahled(obalNahledu, obalProgramu, odkazy, prepinac) {
+
+  class Nahled {
+    constructor(obalNahledu, obalProgramu, prepinac, storage) {
+      this.obalNahledu = obalNahledu
+      this.obalProgramu = obalProgramu
+      this.storage = storage
+      this.prepinac = prepinac
+
+      if (
+        this.viditelny || // ručně zapnuto
+        this.viditelny === null && $(document).width() > 1000 // nenastaveno
+      ) this.zobraz()
+    }
+
+    zobraz() {
+      this.obalProgramu.addClass('programNahled_obalProgramu-zuzeny')
+      this.obalNahledu.addClass('programNahled_obalNahledu-viditelny')
+      this.storage.setItem('viditelny', '1')
+      this.prepinac.html('vypnout náhledy')
+    }
+
+    skryj() {
+      this.obalProgramu.removeClass('programNahled_obalProgramu-zuzeny')
+      this.obalNahledu.removeClass('programNahled_obalNahledu-viditelny')
+      this.storage.setItem('viditelny', '')
+      this.prepinac.html('zapnout náhledy')
+    }
+
+    get viditelny() {
+      return this.storage.getItem('viditelny')
+    }
+
+    zobrazSkryj() {
+      this.viditelny ? this.skryj() : this.zobraz()
+    }
+
+    set data(data) {
+      this.obalNahledu.find('.programNahled_nazev').html(data.nazev)
+      this.obalNahledu.find('.programNahled_stitky').html(data.stitky.map(function(stitek) {
+        return '<div class="programNahled_stitek">'+stitek+'</div>'
+      }).join(''))
+      this.obalNahledu.find('.programNahled_vypraveci').html(data.vypraveci.join(', '))
+      this.obalNahledu.find('.programNahled_kratkyPopis').html(data.kratkyPopis)
+      this.obalNahledu.find('.programNahled_popis').html(data.popis)
+      this.obalNahledu.find('.programNahled_obrazek').attr('src', data.obrazek)
+
+      this.obalNahledu.addClass('programNahled_obalNahledu-maData')
+
+      this.obalNahledu.find('.programNahled_text').scrollTop(0);
+    }
+  }
+
+  class Api {
+    constructor() {
+      this.cache = {}
+    }
+
+    nacti(id) {
+      var api = this
+      if (id in this.cache) {
+        return new Promise(function(resolve) {
+          resolve(api.cache[id])
+        })
+      } else {
+        return $.ajax('program-nahled-api', {data: {idAktivity: id}}).then(function(data) {
+          api.cache[id] = data
+          return data
+        })
+      }
+    }
+  }
+
+  var nahled = new Nahled(
+    obalNahledu,
+    obalProgramu,
+    prepinac,
+    window.localStorage // TODO později přidat nějaký wrapper objekt kvůli prefixování
```

Comment:
Ve zdrojácích Gameconu je už hodně TODOs a mají obvyklý nesmrtelný život - původní autor na ně už nesáhne, protože dělá důležitější věci a nováčci na ně taky raději nesáhnou, protože se bojí, že tím smažou důležitou poznámku.
Nepřidal bys ten wrapper hned?

Reply:
Upravil jsem to (resp. oprefixoval rucne, jak vidim jsou to jenom dve volani). Kdybych psal wrapper rucne, frontendisti by mi urcite omlatili o hlavu, ze jsem mel nainstalovat nejakou knihovnu npmkem nebo yarnem nebo whatever dneska frci :) 

//

Jestli jsem na nic nezapomnel, zmeny by mely byt v commitu, co jsem prihodil. Diky, ze delate ty reviews!

Reply:
Tohle jsem nepochopil, asi proto, že nejsem frontendista:D Co jsem tak narychlo pogooglil, tak asi jde o to, že když se používá localStorage globálně, tak si tam můžou jednotlivé části programu líst do zelí a přepisovat si navzájem hodnoty, takže nějaký prefix klíčů by to vyřešil?
Pokud to TODO bylo kvůli tomu, tak mi připadají dostatečné ty dlouhé názvy, co jsi tam poctivě vymyslel (`programNahled_kdesicosi`), takže za mě to to OK.